engine/api/authentication/session.go

```diff
@@ -12,19 +12,28 @@ import (
 	"github.com/ovh/cds/sdk"
 )
 
-// NewSession returns a new session for a given auth consumer.
-func NewSession(ctx context.Context, db gorpmapper.SqlExecutorWithTx, c *sdk.AuthConsumer, duration time.Duration, mfaEnable bool) (*sdk.AuthSession, error) {
+const (
+	MFADuration = 15 * time.Minute
+)
+
+// NewSession returns a new session for a given auth consumer, and a duration for MFA validity
+func NewSession(ctx context.Context, db gorpmapper.SqlExecutorWithTx, c *sdk.AuthConsumer, duration time.Duration, mfaEnable bool) (*sdk.AuthSession, time.Duration, error) {
 	s := sdk.AuthSession{
 		ConsumerID: c.ID,
 		ExpireAt:   time.Now().Add(duration),
 		MFA:        mfaEnable,
 	}
 
+	var mfaDuration time.Duration
+	if mfaEnable {
+		mfaDuration = MFADuration
+	}
+
 	if err := InsertSession(ctx, db, &s); err != nil {
-		return nil, err
+		return nil, mfaDuration, err
 	}
 
-	return &s, nil
+	return &s, mfaDuration, nil
```

Comment:
Could be nice to return the computed MFAExpireAt to be able to compute both ExpireAt and MFAExpireAt from the same time.Now() value.

Reply:
done